CIP-0102/README.md

```diff
@@ -0,0 +1,177 @@
+
+## Motivation
+
+The inability to create trustless onchain royalty validation with [CIP-0027][] is a major drawback to Cardano NFTs. The pattern defined in CIP-68 represents an opportunity to upgrade the standard to support onchain validation. This CIP aims to eliminate that drawback and demonstrate better support for developers, NFT creators, and NFT collectors, ultimately attracting dapps & NFT projects that would otherwise have taken their talents to another blockchain.
+
+In addition, this standard allows royalties to be split between multiple addresses, another limitation of the CIP-27 royalty schema. Future versions of this standard could  also easily support multiple royalty policies defined for a single collection, applied at the level of individual tokens.
+
+## Specification
+
+### 500 Royalty Datum Standard
+
+The following defines the `500` Royalty NFT standard with the registered `asset_name_label` prefix value
+
+| asset_name_label            | class        | description                                                          |
+| --------------------------- | ------------ | -------------------------------------------------------------------- |
+| 500                         | NFT          | Royalty NFT stored in a UTxO containing a datum with royalty information |
+
+#### Class
+
+The `royalty NFT` is an NFT (non-fungible token).
+
+#### Pattern
+
+The `royalty NFT` **must** have an identical `policy id` as the collection.
+
+The `asset name` **must** be `001f4d70526f79616c7479` (hex encoded), it contains the [CIP-0067][] label `500` followed by the word "Royalty".
+
+Example:\
+`royalty NFT`: `(500)Royalty`\
+`reference NFT`: `(100)Test123`
+
+#### 500 Datum Metadata
+
+The royalty info datum is specified as follows (CDDL):
+
+```cddl
+big_int = int / big_uint / big_nint
+big_uint = #6.2(bounded_bytes)
+big_nint = #6.3(bounded_bytes)
+
+optional_big_int = #6.121([big_int]) / #6.122([])
+
+royalty_recipient = #6.121([
+              address,                    ; definition can be derived from: 
+                                          ; https://github.com/input-output-hk/plutus/blob/master/plutus-ledger-api/src/PlutusLedgerApi/V1/Address.hs#L31
+              int,                        ; variable fee ( calculation: ⌊1 / (fee / 10)⌋ ); integer division with precision 10
+              optional_big_int,           ; min fee (absolute value in lovelace)
+              optional_big_int,           ; max fee (absolute value in lovelace)
+            ])
+
+royalty_recipients = [ * royalty_recipient ]
+
+; version is of type int, we start with version 1
+version = 1  
+
+; Custom user defined plutus data.
+; Setting data is optional, but the field is required
+; and needs to be at least Unit/Void: #6.121([])
+extra = plutus_data
+
+royalty_info = #6.121([royalty_recipients, version, extra])
```

Comment:
Suggestion, bundle the version separately, so that the format can be adjusted if needed without prior knowledge of the version. For example:


```
royalty_info = #6.121([1, royalty_info_v1])
royalty_info_v1 = #6.121([royalty_recipients, extra])
```

Reply:
This is a very interesting idea. If I understand you correctly, this would be helpful for parsing the datum as a third party once multiple versions are introduced. One could parse the version number first, then know what type to expect for the royalty info itself. Is that what you're saying?

I like the idea, but I'm not sure how much of a difference it would actually make. Do you have a good idea of how much more efficient a change like this would be?

My only real concern is compatibility with existing implementations. I'm not sure this provides enough value to warrant making Nebula & other nebula-based implementations incompatible.